File: rust/axum/src/modules/account/controller.rs
use axum::{
    extract::{Path, Query}, http::StatusCode, response::IntoResponse, Extension, Json
};
use validator::Validate;
use crate::{common::errors::bad_request::bad_request_error, modules::auth::interfaces::JwtPayload};
use crate::common::messages::error::ErrorMessages;

use super::service::AccountService;
use super::dto::{
    FindAllQueryAccountDto,
    CreateAccountDto,
    FindOneAccountDto,
};

pub struct AccountController {
}

impl AccountController {
    pub async fn find_all(
        Query(mut dto): Query<FindAllQueryAccountDto>,
    ) -> impl IntoResponse {
        dto.transform();

        if let Err(err) = dto.validate() {
            return bad_request_error(ErrorMessages::BAD_REQUEST, Some(err)).into_response();
        }

        match AccountService::find_all(&dto.cpf).await {
            Ok(accounts) => Json(accounts).into_response(),
            Err(err) => err.into_response(),
        }
    }

    pub async fn create(
        Extension(auth): Extension<JwtPayload>,
        Json(dto): Json<CreateAccountDto>,
    ) -> impl IntoResponse {
        if let Err(err) = dto.validate() {
            return bad_request_error(ErrorMessages::BAD_REQUEST, Some(err)).into_response();
        }

        match AccountService::create(dto, &auth.id, auth.clone()).await {
            Ok(_) => StatusCode::CREATED.into_response(),
            Err(err) => err.into_response(),
        }
    }

    pub async fn find_by_id(
        Extension(auth): Extension<JwtPayload>,
        Path(id): Path<String>,
    ) -> impl IntoResponse {
        let dto = FindOneAccountDto { id };

        if let Err(err) = dto.validate() {
            return bad_request_error(ErrorMessages::BAD_REQUEST, Some(err)).into_response();
        }

        match AccountService::find_by_id(&dto.id, &auth.id, auth.role).await {
            Ok(mut account) => {
                // Omitindo campos não necessários
                account.id_agencia = None;
                account.id_cliente = None;
                Json(account).into_response()
            },
            Err(err) => err.into_response(),
        }
    }
}

File: go/fiber_orm/internal/modules/account/controller.go
package account

import (
	"tcc/internal/common/errors"
	"tcc/internal/common/helper"
	"tcc/internal/common/messages"
	"tcc/internal/modules/account/dto"

	"github.com/gofiber/fiber/v2"
)

type AccountController struct{
	service AccountService
}

func (c *AccountController) FindAll(ctx *fiber.Ctx) error {
	cpf := ctx.Query("cpf")

	dto := dto.FindAllQueryAccountDto{
		CPF: cpf,
	}

	// Transformações Necessárias
	dto.Transform()

	if err := helper.Validate(dto); err != nil {
		badRequest := errors.BadRequestError(messages.ErrorMessages.BadRequest, err...)
		return ctx.Status(badRequest.StatusCode).JSON(badRequest)
	}

	client, err := c.service.FindAll(dto.CPF)
	if err != nil {
		return ctx.Status(err.StatusCode).JSON(err)
	}

	return ctx.Status(fiber.StatusOK).JSON(client)
}

func (c *AccountController) Create(ctx *fiber.Ctx) error {
	auth := helper.GetAuth(ctx)

	var dto dto.CreateAccountDto
	if err := ctx.BodyParser(&dto); err != nil {
		badRequest := errors.BadRequestError(messages.ErrorMessages.BadRequest, err.Error())
		return ctx.Status(badRequest.StatusCode).JSON(badRequest)
	}
	
	if err := helper.Validate(dto); err != nil {
		badRequest := errors.BadRequestError(messages.ErrorMessages.BadRequest, err...)
		return ctx.Status(badRequest.StatusCode).JSON(badRequest)
	}

	_, err := c.service.Create(dto, auth.ID, *auth)
	if err != nil {
		return ctx.Status(err.StatusCode).JSON(err)
	}

	return ctx.Status(fiber.StatusCreated).Send(nil)
}

func (c *AccountController) FindByCPF(ctx *fiber.Ctx) error {
	id := ctx.Params("id")

	auth := helper.GetAuth(ctx)

	dto := dto.FindOneAccountDto{
		ID: id,
	}

	if err := helper.Validate(dto); err != nil {
		badRequest := errors.BadRequestError(messages.ErrorMessages.BadRequest, err...)
		return ctx.Status(badRequest.StatusCode).JSON(badRequest)
	}

	account, err := c.service.FindById(id, auth.ID, auth.Role)
	if err != nil {
		return ctx.Status(err.StatusCode).JSON(err)
	}

	// Omitindo Campos não necessários
	account.IDAgencia = nil
	account.IDCliente = nil

	return ctx.Status(fiber.StatusOK).JSON(account)
}